### PR TITLE
Add User list functionality

### PR DESCRIPTION
**Pull request by emanuelcasco**
## Summary

- Add User list functionality:
  - Route, controller and service.
  - User can customize pagination with `page` and `limit` params.
- Refactor in validations middleware to review either query or body.
- Add tests to `GET /users`.

## Known Issues

None

## Trello Card

- https://trello.com/c/IPwB5oBc/4-lista-de-usuarios

### CODE REVIEW

File: app/services/users.js [hunk 1]
```diff
@@ -17,3 +17,15 @@ exports.updateOrCreate = record =>
...
+
+exports.getAll = ({ limit = 5, page = 0 }) =>
+  User.findAndCountAll({
+    attributes: ['firstName', 'lastName', 'email', 'isAdmin'],
+    limit,
+    offset: page * limit,
+    order: [['id', 'ASC']]
+  })
+    .then(data => Object.assign({ page }, data))
+    .catch(err => {
```
Comment (darkBasic666) [inline app/services/users.js]: log error

**Comment by darkBasic666** [comment]
Log errors in services and controller.

File: test/users.test.js [hunk 2]
```diff
@@ -358,5 +358,62 @@ describe('users', () => {
         })
         .then(() => done());
     });
+    describe('/users GET', () => {
```
Comment (darkBasic666) [inline test/users.test.js]: Add tests to queries params.

File: app/services/users.js (hunk 1 above)
Comment (emanuelcasco) [reply app/services/users.js]: It is necessary to log error here? If it throws an error then express' errors middleware should handle it. Then, error middleware will log error and send the response to user. Correct me if I misunderstood something 😋.

**Pull request merged**